app/controllers/collections_controller.rb

```diff
@@ -475,4 +542,94 @@ def format_update_collection_metadata(collection, edited_metadata, overwrite)
     new_metadata
   end
 
+  # Returns an Alveo formatted collection full URL
+  def format_collection_url(collection_name)
+    collection_url(collection_name)
+  end
+
+  # Returns an Alevo formatted item full URL
+  def format_item_url(collection_name, item_name)
+    catalog_url(collection_name, item_name)
+  end
+
+  # Retuns an Alveo formatted document full URL
+  def format_document_url(collection_name, item_name, document_name)
+    catalog_document_url(collection_name, item_name, document_name)
+  end
+
+  # Overrides the jsonld is_part_of_corpus with the collection's Alveo url
+  def override_is_part_of_corpus(item_json_ld, collection_name)
+    item_json_ld["@graph"].each do |node|
+      is_doc = node["@type"] == MetadataHelper::DOCUMENT.to_s || node["@type"] == MetadataHelper::FOAF_DOCUMENT.to_s
+      unless is_doc
+        ["dcterms:isPartOf", MetadataHelper::IS_PART_OF.to_s].each do |is_part_of|
+          node[is_part_of]["@id"] = format_collection_url(collection_name) if node.has_key?(is_part_of)
+        end
+      end
+    end
+    item_json_ld
+  end
+
+  # Updates the @ids found in the JSON-LD to be the alveo catalog url
+  def update_ids_in_jsonld(jsonld_metadata, collection)
+    # NOTE: it is assumed that the metadata will contain only items at the outermost level and documents nested within them
+    jsonld_metadata["@graph"].each do |item_metadata|
+      # Update item @ids
+      item_id = get_dc_identifier(item_metadata)
+      item_metadata = update_jsonld_item_id(item_metadata, collection.name)
+      # Update document @ids
+      ['ausnc:document', MetadataHelper::DOCUMENT.to_s].each do |doc_predicate|
+        if item_metadata.has_key?(doc_predicate)
+          if item_metadata[doc_predicate].is_a? Array
+            # When "ausnc:document" contains an array of document hashes
+            item_metadata[doc_predicate].each do |document_metadata|
+              document_metadata = update_jsonld_document_id(document_metadata, collection.name, item_id)
+            end
+          else
+            # When "ausnc:document" contains a single document hash
+            item_metadata[doc_predicate] = update_jsonld_document_id(item_metadata[doc_predicate], collection.name, item_id)
+          end
+        end
+      end
+      # Update display document @ids
+      ["hcsvlab:display_document", MetadataHelper::DISPLAY_DOCUMENT.to_s].each do |display_doc|
```

Comment:
It looks like this loop is has the exact same logic as the one below. Could you just merge them? i.e.
```
["hcsvlab:display_document", MetadataHelper::DISPLAY_DOCUMENT.to_s, "hcsvlab:indexable_document", MetadataHelper::INDEXABLE_DOCUMENT].each do |document|
```